cmd/kube-controllers/main.go

```diff
@@ -295,12 +296,25 @@ func newEtcdV3Client() (*clientv3.Client, error) {
 		return nil, err
 	}
 
+	if config.Spec.EtcdEndpoints != "" && config.Spec.EtcdDiscoverySrv != "" {
```

Comment:
The [equivalent code in libcalico-go](https://github.com/jsok/libcalico-go/blob/e24e921bebd2fd934d3d88d86f98887488b87050/lib/backend/etcdv3/etcdv3.go#L48) can't be re-used in it's current state as it doesn't expose the `etcd.clientv3.Client` client embedded in `libcalico-go/lib/backend/api.Client`. Have had to duplicate it instead.